src/main/user.js

```diff
@@ -24,12 +17,21 @@ class User {
     return lodgingCost + flightCost;
   }
 
+  findDestinationDetails(trip) {
+    let destination = this.destinationKey
+      .find(destination => trip.destinationID === destination.id);
+    destination.tripID = trip.id;
+    return destination;
+  }
+	
   calculateTripCost4Yr(year) {
     if (!year.match(/\d{4}/)) {
       return 'Invalid Input';
     }
 
-    return this.approvedTrips
+    let approvedTrips = this.trips.filter(trip => trip.status === "approved")
```

Comment:
This is smart, I hadn't thought about how the pending trips shouldn't count towards the total spent.